llvm/include/llvm/Analysis/DebugInfoCache.h

```diff
@@ -0,0 +1,43 @@
+//===- llvm/Analysis/DebugInfoCache.h - debug info cache ---*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file contains an analysis that builds a cache of debug info for each
+// DICompileUnit in a module.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_ANALYSIS_DEBUGINFOCACHE_H
+#define LLVM_ANALYSIS_DEBUGINFOCACHE_H
+
+#include "llvm/IR/DebugInfo.h"
+#include "llvm/IR/PassManager.h"
+
+namespace llvm {
+class DebugInfoCache {
```

Comment:
Can you add doxygen comments explaining what exactly is being cached and what the use-case for this is?

Reply:
Thanks @adrian-prantl, will do!